### PR TITLE
Update ffmpeg version

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Change to nasm build (yasm is no longer developed and ffmpeg page shows nasm as the dependency) and update ffmpeg version.

### CODE REVIEW

**Comment by adamjstewart** [comment]
It looks like a tarball got sucked into this PR.

File: var/spack/repos/builtin/packages/darshan-runtime/package.py [hunk 1]
```diff
@@ -34,9 +34,9 @@ class DarshanRuntime(Package):
     systems where you intend to instrument MPI applications."""
 
     homepage = "http://www.mcs.anl.gov/research/projects/darshan/"
-    url = "http://ftp.mcs.anl.gov/pub/darshan/releases/darshan-3.1.0.tar.gz"
+    url = "https://xgitlab.cels.anl.gov/darshan/darshan/repository/darshan-3.1.6/archive.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/darshan-runtime/package.py]: Why the URL change? Have any of the old checksums changed as well?

File: var/spack/repos/builtin/packages/ffmpeg/package.py [hunk 2]
```diff
@@ -47,3 +48,4 @@ def configure_args(self):
             config_args.append('--enable-shared')
 
         return config_args
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ffmpeg/package.py]: This line will probably cause flake8 to fail.

File: var/spack/repos/builtin/packages/darshan-runtime/package.py (hunk 1 above)
Comment (s-sajid-ali) [reply var/spack/repos/builtin/packages/darshan-runtime/package.py]: It wasn't working as explained [here](https://www.mcs.anl.gov/research/projects/darshan/2018/03/29/ftp-download-problems/)

**Comment by s-sajid-ali** [comment]
I'll see if I can make a new branch from develop and paste the package.py for ffmpeg there.

File: var/spack/repos/builtin/packages/darshan-runtime/package.py (hunk 1 above)
Comment (s-sajid-ali) [reply var/spack/repos/builtin/packages/darshan-runtime/package.py]: Also, the mcs site is very unreliable half the time in my experience.

**Comment by adamjstewart** [comment]
No need for a new branch, just add a second commit to this branch.

**Comment by s-sajid-ali** [comment]
The tests fail :
```
[sajid@xrmlite spack]$ spack install --test=root ffmpeg
==> nasm is already installed in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/nasm-2.13.03-exjmvybje2tdyzh35x3jkbvyamq2jl73
==> Installing ffmpeg
==> Using cached archive: /home/sajid/packages/spack/var/spack/cache/ffmpeg/ffmpeg-4.0.1.tar.bz2
==> Staging archive: /home/sajid/packages/spack/var/spack/stage/ffmpeg-4.0.1-uucox2v5y5w3t5evcrozrneq576rvwfi/ffmpeg-4.0.1.tar.bz2
==> Created stage in /home/sajid/packages/spack/var/spack/stage/ffmpeg-4.0.1-uucox2v5y5w3t5evcrozrneq576rvwfi
==> No patches needed for ffmpeg
==> Building ffmpeg [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j32' 'check'

3 errors found in build log:
     3661    CC tests/api/api-flac-test.o
     3662    CC tests/api/api-seek-test.o
     3663    HOSTCC     tests/audiogen.o
     3664    HOSTCC     tests/videogen.o
     3665    GEN        tests/data/hls-list.m3u8
     3666    GEN        tests/data/hls-list-append.m3u8
  >> 3667    make: *** [tests/data/hls-list.m3u8] Error 127
     3668    make: *** Waiting for unfinished jobs....
  >> 3669    make: *** [tests/data/hls-list-append.m3u8] Error 127
     3670    GEN        tests/data/add_keyframe_index.flv
  >> 3671    make: *** [tests/data/add_keyframe_index.flv] Error 127
     3672    tests/api/api-flac-test.c: In function ‘run_test’:
     3673    tests/api/api-flac-test.c:166:9: warning: ‘avcodec_encode_audio2’ is deprecated [-Wdeprecated-declarations]
     3674             result = avcodec_encode_audio2(enc_ctx, &enc_pkt, in_frame, &got_output);
     3675             ^~~~~~
     3676    In file included from tests/api/api-flac-test.c:30:0:
     3677    ./libavcodec/avcodec.h:5347:5: note: declared here

See build log for details:
  /home/sajid/packages/spack/var/spack/stage/ffmpeg-4.0.1-uucox2v5y5w3t5evcrozrneq576rvwfi/ffmpeg-4.0.1/spack-build.out
```

**Comment by adamjstewart** [comment]
Can you upload your build log? I'm not seeing anything that makes it clear what the problem is.

**Comment by s-sajid-ali** [comment]
[Build log](https://pastebin.com/hJPC2k3i).

**Comment by chuckatkins** [comment]
> Updating darshan in this pr seems unrelated and out of scope for an ffmpeg update. Additionally, the darshan packages have already been updated in more recent PRs. Please remove the darshan update then I'm happy to merge this.

I take that back.  ffmpeg has already been updated upstream so the changes for this PR are obsolete.  Given that, I'm going to go ahead and close this.  @s-sajid-ali if there's a compelling reason to do so and you'd like to open a new pr to switch the asm dependency yasm to nasm, then please feel free to do so and add me as a reviewer.

**Pull request closed without merging**